### PR TITLE
PR: Add validation for empty signature while getting calltip in the IPython Console

### LINKED ISSUE #9570: Error for calltip with empty signature
### What steps will reproduce the problem?

import os

os.getcwd( -> triggers calltip but an error raises

### Traceback
```python-traceback
  File "C:\Users\Daniel\Anaconda3\envs\spyder4\lib\site-packages\qtconsole\base_frontend_mixin.py", line 138, in _dispatch
    handler(msg)
  File "C:\Users\Daniel\Desktop\Spyder\spyder\spyder\plugins\ipythonconsole\widgets\help.py", line 141, in _handle_inspect_reply
    documentation = self.get_documentation(content)
  File "C:\Users\Daniel\Desktop\Spyder\spyder\spyder\plugins\ipythonconsole\widgets\help.py", line 46, in get_documentation
    documentation = text.split(signature)
ValueError: empty separator
```

## Versions

* Spyder version: 4.0.0.dev0 eb70e6b6d
* Python version: 3.7.3
* Qt version: 5.9.6
* PyQt5 version: 5.9.2
* Operating System: Windows 8.1

### Dependencies

```
pygments >=2.0     :  2.4.0 (OK)
qdarkstyle >=2.6.4 :  2.6.8 (OK)
sphinx >=0.6.6     :  2.0.1 (OK)
pyls >=0.19.0;<0.25:  0.24.0 (OK)
nbconvert >=4.0    :  5.5.0 (OK)
pandas >=0.13.1    :  0.24.2 (OK)
numpy >=1.7        :  1.16.3 (OK)
sympy >=0.7.3      :  1.4 (OK)
cython >=0.21      :  0.29.7 (OK)
qtconsole >=4.2.0  :  4.5.1 (OK)
IPython >=4.0      :  7.5.0 (OK)
matplotlib >=2.0.0 :  3.0.3 (OK)
pylint >=0.25      :  2.3.1 (OK)
```

### PR DESCRIPTION
**Pull request by dalthviz**
## Description of Changes

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)

* [ ] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

### Issue(s) Resolved

Fixes #9570 

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: dalthviz

### CODE REVIEW

File: spyder/plugins/ipythonconsole/widgets/help.py [hunk 1]
```diff
@@ -43,9 +43,10 @@ def get_documentation(self, content):
         text = data.get('text/plain', '')
         signature = self.get_signature(content).split('(')[-1]
         if text:
-            documentation = text.split(signature)
-            if len(documentation) > 1:
-                return documentation[-1]
+            if signature:
+                documentation = text.split(signature)
+                if len(documentation) > 1:
+                    return documentation[-1]
```
Comment (ccordoba12) [inline spyder/plugins/ipythonconsole/widgets/help.py]: I think this would avoid showing documentation for objects without signature, right? I mean, if there's no signature we should still show the documentation (if `text` has some content).

**Comment by ccordoba12** [comment]
@dalthviz, as part of this PR please also update our qtconsole requirement here:

https://github.com/spyder-ide/spyder/blob/master/spyder/plugins/ipythonconsole/plugin.py#L69

I forgot to mention that in your previous PR, sorry.

File: spyder/plugins/ipythonconsole/widgets/help.py [hunk 2]
```diff
@@ -139,10 +147,9 @@ def _handle_inspect_reply(self, rep):
...
-                        signature,
-                        documentation=documentation,
-                        language=self.language_name,
-                        max_lines=5
-                    )
+                self._control.show_calltip(
+                    signature,
+                    documentation=documentation,
+                    language=self.language_name,
+                    max_lines=5
```
Comment (ccordoba12) [inline spyder/plugins/ipythonconsole/widgets/help.py]: Let's change this to `7` because several docstrings are cutoff at the `Parameters` line.

File: spyder/plugins/ipythonconsole/widgets/help.py [hunk 3]
```diff
@@ -38,14 +38,20 @@ def clean_invalid_var_chars(self, var):
...
-        """Get documentation fron inspect reply content."""
+        """Get documentation from inspect reply content."""
         data = content.get('data', {})
         text = data.get('text/plain', '')
-        signature = self.get_signature(content).split('(')[-1]
         if text:
-            documentation = text.split(signature)
-            if len(documentation) > 1:
-                return documentation[-1]
+            text = re.compile(ANSI_PATTERN).sub('', text)
```
Comment (ccordoba12) [inline spyder/plugins/ipythonconsole/widgets/help.py]: Why is this necessary?

Comment (dalthviz) [reply spyder/plugins/ipythonconsole/widgets/help.py]: This is needed to remove characters like `[0m` but no characters like `new line` (`\n`):

without_p

Comment (ccordoba12) [reply spyder/plugins/ipythonconsole/widgets/help.py]: Ok, thanks for the explanation.
